utopia-remix/app/util/api.server.ts

```diff
@@ -142,3 +165,13 @@ export async function requireUser(
     throw error
   }
 }
+
+export async function getUser(request: Request): Promise<UserDetails | null> {
```

Comment:
can you please use this function inside `requireUser`? this is a copy of the code over there, and it's _very_ important we don't have diverging behaviors for this stuff

Reply:
I unified the session id fetching logic, but `requireUser` and `getUser` have slightly different behaviors (`requireUser` throws an error if there is an empty session id or no user, while `getUser` only returns null and it's ok)